src/libaktualizr/uptane/imagesrepository.cc

```diff
@@ -65,54 +63,55 @@ bool ImagesRepository::verifySnapshot(const std::string& snapshot_raw) {
   return true;
 }
 
-bool ImagesRepository::verifyTargets(const std::string& targets_raw, const Uptane::Role& role) {
-  try {
-    const Json::Value targets_json = Utils::parseJSON(targets_raw);
-    const std::string canonical = Utils::jsonToCanonicalStr(targets_json);
-    // Delegated targets do not have their hashes stored in their parent.
-    if (role == Uptane::Role::Targets()) {
-      bool hash_exists = false;
-      for (const auto& it : snapshot.targets_hashes()) {
-        switch (it.type()) {
-          case Hash::Type::kSha256:
-            if (Hash(Hash::Type::kSha256, boost::algorithm::hex(Crypto::sha256digest(canonical))) != it) {
-              LOG_ERROR << "Hash verification for targets metadata failed";
-              return false;
-            }
-            hash_exists = true;
-            break;
-          case Hash::Type::kSha512:
-            if (Hash(Hash::Type::kSha512, boost::algorithm::hex(Crypto::sha512digest(canonical))) != it) {
-              LOG_ERROR << "Hash verification for targets metadata failed";
-              return false;
-            }
-            hash_exists = true;
-            break;
-          default:
-            break;
+bool ImagesRepository::verifyRoleHashes(const std::string& role_data, const Uptane::Role& role) const {
+  const std::string canonical = Utils::jsonToCanonicalStr(Utils::parseJSON(role_data));
+  bool hash_exists = false;
+  for (const auto& it : snapshot.role_hashes(role)) {
+    switch (it.type()) {
+      case Hash::Type::kSha256:
+        if (Hash(Hash::Type::kSha256, boost::algorithm::hex(Crypto::sha256digest(canonical))) != it) {
+          LOG_ERROR << "Hash verification for " << role.ToString() << " metadata failed";
+          return false;
         }
-      }
-      if (!hash_exists) {
-        LOG_ERROR << "No hash found for targets.json";
-        return false;
-      }
+        hash_exists = true;
+        break;
+      case Hash::Type::kSha512:
+        if (Hash(Hash::Type::kSha512, boost::algorithm::hex(Crypto::sha512digest(canonical))) != it) {
+          LOG_ERROR << "Hash verification for " << role.ToString() << " metadata failed";
+          return false;
+        }
+        hash_exists = true;
+        break;
+      default:
+        break;
```

Comment:
Looks like my comment may have disappeared. Is it possible for the other `verify` functions to reuse `verifyRoleHashes` as well? If it's too complicated, don't worry about it.

Reply:
`verifyTargets` is already using it. Other roles are not listed in `snapshot.json` (well, `root.json` is, but it makes no sense to verify it).

Reply:
I meant specifically the hashing logic in `verifySnapshot`. But after looking more closely, it doesn't isn't worth trying to combine those.

Reply:
@patrickvacek Yes, maybe the logic of iterating over a list of hashes could be abstracted out a bit, but right now it's not too bad I think, it's just two places.